octree/include/pcl/octree/impl/octree_pointcloud.hpp

```diff
@@ -666,7 +712,14 @@ pcl::octree::OctreePointCloud<PointT, LeafContainerT, BranchContainerT, OctreeT>
     getPointByIndex(const uindex_t index_arg) const
 {
   // retrieve point from input cloud
-  assert(index_arg < input_->size());
+  if (index_arg >= input_->size()) {
+    PCL_ERROR("[pcl::octree::OctreePointCloud::getPointByIndex] Input point index "
+              "(%lu) must be < input cloud size (%lu)\n",
+              index_arg,
+              input_->size());
+    PointT point;
+    return point;
```

Comment:
Here's `error: reference to local variable 'point' returned [-Werror=return-local-addr]`, maybe we should redesign the API to `bool getPointByIndex(const uindex_t index_arg, PointT&) const`?